docs/general/ddl/data-types.rst

```diff
@@ -303,1283 +458,2517 @@ Casting from ``OBJECT`` to ``JSON``::
+much slower compared to operations on the integer or floating-point types.
 
-.. csv-table::
-   :header: "Expression", "Return Type", "Description"
+The ``NUMERIC`` type can be configured with the ``precision`` and
+``scale``. The ``precision`` value of a numeric is the total count of
+significant digits in the unscaled numeric value. The ``scale`` value of a
+numeric is the count of decimal digits in the fractional part, to the right of
+the decimal point. For example, the number 123.45 has a precision of ``5`` and
+a scale of ``2``. Integers have a scale of zero.
 
-   "timestamp without time zone AT TIME ZONE zone", "timestamp with time zone", "Treat \
-   given time stamp without time zone as located in the specified time zone"
-   "timestamp with time zone AT TIME ZONE zone", "timestamp without time zone", "Convert \
-   given time stamp with time zone to the new time zone, with no time zone designation"
+To declare the ``NUMERIC`` type with the precision and scale, use the syntax::
 
-In these :ref:`expressions <gloss-expression>`, the desired time zone is
-specified as a string (e.g., 'Europe/Madrid', '+02:00').
+    NUMERIC(precision, scale)
 
-The scalar function :ref:`TIMEZONE <scalar-timezone>` (zone, timestamp) is
-equivalent to the SQL-conforming timestamp construct ``AT TIME ZONE zone``.
+Alternatively, only the precision can be specified, the scale will be zero
+or positive integer in this case::
 
-.. _time-data-type:
+    NUMERIC(precision)
 
-time with time zone
--------------------
+Without configuring the precision and scale the ``NUMERIC`` type value will be
+represented by an unscaled value of the unlimited precision::
 
-The time type consists of time followed by an optional time zone.
+    NUMERIC
 
-``timetz`` is an alias for `time with time zone`.
+The ``NUMERIC`` type backed internally by the Java ``BigDecimal`` class. For
+more detailed information about its behaviour, see `BigDecimal documentation`_.
 
-`time with time zone` literals can be constructed using a string literal
-and a cast. The syntax for string literal is as follows:
 
-.. code-block:: text
+.. _type-real:
 
-    time-element [offset]
+``REAL``
+''''''''
 
-    time-element: time-only [fraction]
-    time-only:    HH[[:][mm[:]ss]]
-    fraction:     '.' digit+
-    offset:       {+ | -} time-only | geo-region
-    geo-region:   As defined by ISO 8601.
+An inexact `single-precision floating-point`_ value.
 
+Limited to four bytes, with six to nine decimal digits precision.
 
-Where ``time-only`` can contain optional seconds, or optional minutes and
-seconds, and can use ``:`` as a separator optionally.
+Example:
 
-`fraction` accepts up to 6 digits, as precision is in micro seconds.
+::
 
-Time zone syntax as defined by `ISO 8601 time zone designators`_.
+    cr> CREATE TABLE my_table (
+    ...     number REAL
+    ... );
+    CREATE OK, 1 row affected (... sec)
 
-.. NOTE::
+::
 
-    This type cannot be used in `CREATE TABLE` or `ALTER` statements.
```

Comment:
@mkleen and @seut: Does _This type cannot be used in `CREATE TABLE` or `ALTER` statements._ still hold true for `TIME WITH TIME ZONE` aka. `TIMETZ` types (see original documentation at [1])? If so, I should probably bring back this admonition, which has then been accidentally lost.

[1] https://crate.io/docs/crate/reference/en/4.6/general/ddl/data-types.html#time-with-time-zone

Reply:
@amotl Yes it still holds true,`TIME` and `DATE` types cannot be used as storage types.

Reply:
All right, fixed/improved with 559669cfe.